server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -431,13 +438,35 @@ private void executeRequest(
         if (originalSearchRequest.isPhaseTook() == null) {
             originalSearchRequest.setPhaseTook(clusterService.getClusterSettings().get(SEARCH_PHASE_TOOK_ENABLED));
         }
-        SearchRequestOperationsListener.CompositeListener requestOperationsListeners = searchRequestOperationsCompositeListenerFactory
-            .buildCompositeListener(originalSearchRequest, logger);
+        SearchRequestOperationsListener.CompositeListener requestOperationsListeners;
```

Comment:
This issue that the instrumentation does not work here they way it should is because we never completed [1], this pull request have clear dependency on transport action instrumentation. I think this is what we can do here, wrap this action execution with (you could borrow `SpanBuilder.from` from [1]`):
```
final Span span = tracer.startSpan(SpanBuilder.from(task, actionName));
try (final SpanScope spanScope = tracer.withSpanInScope(span)) {
 .... 
}
```

It will give you the span to stick to.


[1] https://github.com/opensearch-project/OpenSearch/pull/10096/files

Reply:
Are you recommending to create the span here in TransportSearchAction instead of traceableSearchRequestOperationsListener onRequestStart()?

Reply:
> Are you recommending to create the span here in TransportSearchAction 

Create parent span here (that would be action), the TraceableSearchRequestOperationsListener  would use this span as a parent

Reply:
This is done now. Noting that this SpanScope encompasses the rest of the `executeRequest`